cmd/ci-operator-prowgen/main.go

```diff
@@ -386,7 +400,11 @@ func generateJobs(
 		}
 
 		presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest("images", repoInfo, generatePodSpec(repoInfo.configFilename, "[images]", additionalPresubmitArgs...)))
-		postsubmits[orgrepo] = append(postsubmits[orgrepo], *generatePostsubmitForTest("images", repoInfo, true, labels, generatePodSpec(repoInfo.configFilename, "[images]", additionalPostsubmitArgs...)))
+
+		// If we have and explicit promotion config, let's respect that. Otherwise, validate if the branch matches promotion target
+		if configSpec.PromotionConfiguration != nil || shouldBePromoted(repoInfo.branch, promotionNamespace, promotionName) {
```

Comment:
I would go so far as to say this should be `isPromotionBranch` and allow only `release-X.Y` and `master` to be promotion branches.

Reply:
Yeah, probably even better